src/main/java/com/google/jenkins/plugins/k8sengine/CredentialsUtil.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jenkins.plugins.k8sengine;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotCredentials;
+import com.google.jenkins.plugins.k8sengine.client.ContainerScopeRequirement;
+import com.google.jenkins.plugins.k8sengine.client.Messages;
+import hudson.AbortException;
+import hudson.model.ItemGroup;
+import hudson.security.ACL;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import jenkins.model.Jenkins;
+
+/** Provides a library of utility functions for credentials-related work. */
+public class CredentialsUtil {
```

Comment:
Looks great. Just need to add unit tests for this now.

Reply:
Added.